pkg/resource/acl/custom_set_output.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package acl
+
+import (
+	"context"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/memorydb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	svcapitypes "github.com/aws-controllers-k8s/memorydb-controller/apis/v1alpha1"
+)
+
+const (
+	// The number of minutes worth of events to retrieve.
+	// 14 days in minutes
+	eventsDuration = 20160
+)
+
+func (rm *resourceManager) CustomDescribeACLsSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.DescribeACLsOutput,
+	ko *svcapitypes.ACL,
+) (*svcapitypes.ACL, error) {
+	if len(resp.ACLs) == 0 {
+		return ko, nil
+	}
+	elem := resp.ACLs[0]
+	err := rm.customSetOutputSupplementAPIs(ctx, r, elem, ko)
+	if err != nil {
+		return nil, err
+	}
+	return ko, nil
+}
+
+func (rm *resourceManager) customSetOutputSupplementAPIs(
+	ctx context.Context,
+	r *resource,
+	acl *svcsdk.ACL,
+	ko *svcapitypes.ACL,
+) error {
+	events, err := rm.provideEvents(ctx, r.ko.Spec.Name, 20)
+	if err != nil {
+		return err
+	}
+	ko.Status.Events = events
+	return nil
+}
+
+func (rm *resourceManager) provideEvents(
+	ctx context.Context,
+	aclName *string,
+	maxResults int64,
+) ([]*svcapitypes.Event, error) {
+	input := &svcsdk.DescribeEventsInput{}
+	input.SetSourceType("acl")
+	input.SetSourceName(*aclName)
+	input.SetMaxResults(maxResults)
+	input.SetDuration(eventsDuration)
+	resp, err := rm.sdkapi.DescribeEventsWithContext(ctx, input)
+	rm.metrics.RecordAPICall("READ_MANY", "DescribeEvents-ACL", err)
+	if err != nil {
+		rm.log.V(1).Info("Error during DescribeEvents-ACL", "error", err)
+		return nil, err
+	}
+	events := []*svcapitypes.Event{}
+	if resp.Events != nil {
+		for _, respEvent := range resp.Events {
```

Comment:
I believe `range` still works if `resp.Events` is `nil`, we don't need the nil check.

Reply:
Deleted `if resp.Events != nil`